plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java

```diff
@@ -266,6 +341,116 @@ public String getOutputDir(String projectUri) {
     }
   }
 
+  private List<JarDto> getProjectExternalLibraries(ExternalLibrariesParameters params) {
```

Comment:
I guess, all methods below can invoke some common method. It  will simplify the code.
```
private <T, P> List<T> doGet(P params) {
...
  }
```

Reply:
No, they cannot. gson cannot determine the collection type from a type parameter. What did you have in mind, @tolusha ?